aggregator/src/binary_utils.rs

```diff
@@ -302,6 +303,11 @@ where
 
     register_database_pool_status_metrics(pool, &meter)?;
 
+    // Configure the global rayon threadpool, and provide thread names.
+    ThreadPoolBuilder::new()
+        .thread_name(|i| format!("rayon-{i}"))
+        .build_global()?;
```

Comment:
This will spin up all the threads and wait for them to report readiness. Looking at a profile, I think this takes around 300-500 microseconds on my machine.

I added this for two reasons, based on looking at a profile. Firstly, if we don't call `build_global()`, the global threadpool will be instantiated upon its first use. Thus, we may as well front-load this work during initialization. Secondly, Rayon doesn't change the threads' names by default, so they just show up as `janus_aggregator`. Giving them a distinctive name will make future analysis easier.

Reply:
Front loading threadpool creation makes sense, but I'm concerned about paying the cost of doing so (in startup time and in threads) in the case of the aggregation job creator and collection job driver, which don't (yet) use the threadpool. Could we possible conditionalize the threadpool creation based on which process we start?